java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -656,8 +684,11 @@ public static Table readParquet(ParquetOptions opts, HostMemoryBuffer buffer,
    * @return the file parsed as a table on the GPU.
    */
   public static Table readORC(ORCOptions opts, File path) {
-    return new Table(gdfReadORC(opts.getIncludeColumnNames(),
-        path.getAbsolutePath(), 0, 0, opts.usingNumPyTypes(), opts.timeUnit().getNativeId()));
+    long amount = Math.max(path.length(), opts.getRowGuess() * 100);
```

Comment:
Why `* 100` here?  I would expect the schema to be examined to compute the row-size estimate.  If the schema is all fixed-width types then we know how big it should be.  As it is now, it could be wildly off in either direction if the row count is high enough and the user can't easily compensate.  There's no way to override this via a config or other dynamic setting.

If we go with the total size estimate rather than row count estimate approach this isn't an issue.

Reply:
The issue is that there is no schema passed into the ORC reader.  i went with row count because it was something again that an end user might reasonable be able to come up with. And like I said if you want me to change it I can, but I doubt it will produce a more accurate result in the long term.

Reply:
After thinking it over I'll update the APIs to always take a size guess, that if provided we will use.  I'll also provide one, or possibly more APIs to help with the guess based off of row count and schema.